src/ipa/rpi/controller/rpi/clock_recovery.h

```diff
@@ -0,0 +1,43 @@
+#include <chrono>
+#include <list>
+using namespace std::chrono_literals;
+
+namespace RPiController {
+
+class ClockRecovery
+{
+public:	
+	std::list<int64_t> error_values;
+	std::list<int64_t> client_sequence;
+	unsigned int syncPeriod = 30;
+	unsigned int listSize = 100;
+	void updating_values(std::chrono::microseconds delta);
+	std::chrono::microseconds trending_error(std::chrono::microseconds LastWallClock, std::chrono::microseconds ClientWallClock, std::chrono::microseconds lastPayloadFrameDuration, unsigned int sequence);
+	int64_t modeled_wall_clock(int64_t WallClock, int64_t KernelWallClock, unsigned int sequence);
+	void clear();
+	
+	ClockRecovery(unsigned int syncperiod, unsigned int listsize)
+	{
+		syncPeriod = syncperiod;
+		listSize = listsize;
+	}
+	
+	ClockRecovery() {}
+		
+	ClockRecovery(int64_t wallClock, int64_t kernel, unsigned int syncperiod, unsigned int listsize)
+	{
+		wallClock_baseline = wallClock;
+		kernel_baseline = kernel;
+		syncPeriod = syncperiod;
+		listSize = listsize;
+	}
+private:
+	float slope();
+	double xsum = 0;
+	double ysum = 0;
+	double xysum = 0;
+	double x2sum = 0;
+	int64_t wallClock_baseline = 0;
+	int64_t kernel_baseline = 0;
```

Comment:
There's a convention that all member variables should end with `_` (an underscore). We should probably change these (sorry!).